src/common/base/abstractPromote.ts

```diff
@@ -73,77 +58,45 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
     });
     // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
     this.flags = flags as Flags<T>;
-    this.outputService = new OutputServiceFactory().forDeployment(
-      this.flags,
-      (this.flags['devops-center-username'] as Org).getConnection()
+    this.setOutputService(
+      new OutputServiceFactory().forDeployment(
+        this.flags,
+        (this.flags['devops-center-username'] as Org).getConnection()
+      )
     );
+    this.targetOrg = this.flags['devops-center-username'] as Org;
```

Comment:
Maybe move this before **this.setOutputService** and use **this.targetOrg.getConnection()** ??
So you don’t have to cast it twice

Reply:
Good idea. I will change it.